libs/core/src/lib/select/select.component.html

```diff
@@ -12,9 +12,13 @@
             <button class="fd-select-button-custom"
                     fd-button
                     aria-haspopup="true"
-                    [ngClass]="{'fd-button--compact': compact}"
                     [attr.aria-expanded]="isOpen"
-                    [disabled]="disabled">
+                    [disabled]="disabled"
+                    [ngClass]="{'fd-button--compact': compact}"
```

Comment:
I have outlined the syntax for you, why did you use different one.  this is not how its done. Please see bellow
```html
<button ..
      [ngClass]="{
           'fd-button--compact':  compact,
           'fd-button--noborder':  selectType === 'noborder',
           'fd-button--splitborder':  selectType === 'splitborder', 
       }" >

```